### PR TITLE
add deprecation message to `google_notebook_instance`

### PR DESCRIPTION
**Pull request by BBBmau**
Adds a deprecation message to `google_notebook_instance`

Fixes https://github.com/hashicorp/terraform-provider-google/issues/17187

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by bcreddy-gcp** [comment]
When will the resource be removed from Terraform? From notebooks end, we plan to support the resource for the next year, after which it will still be available but not be supported. We'd prefer the resource being in Terraform for as long instances are available.

**Comment by BBBmau** [comment]
@bcreddy-gcp no plans to remove it anytime soon, it would be good practice to mark it as deprecated so that users can shift to using `google_workbench_instance` instead based off of your comment left from another issue.

https://github.com/hashicorp/terraform-provider-google/issues/12515#issuecomment-1927822049

**Comment by c2thorn** [comment]
> When will the resource be removed from Terraform? From notebooks end, we plan to support the resource for the next year, after which it will still be available but not be supported. We'd prefer the resource being in Terraform for as long instances are available.

Can we capture this in a comment right above the deprecation message? @BBBmau 
There is a non-zero chance that as we plan the next major release, someone goes through deprecated resources and marks them for removal. I want to make sure this doesn't get accidentally marked until appropriate

File: mmv1/products/notebooks/Instance.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@
 
 --- !ruby/object:Api::Resource
 name: 'Instance'
+deprecation_message: >-
```
Comment (c2thorn) [inline mmv1/products/notebooks/Instance.yaml:16]:
```suggestion
# This resource should not be removed until the 2025 major release or later. 
# Check instance availability first before fully removing.
deprecation_message: >-
```

Comment (BBBmau) [reply mmv1/products/notebooks/Instance.yaml:16]: Thanks for the comment suggestion!

Comment (c2thorn) [reply mmv1/products/notebooks/Instance.yaml:18]: sorry about the whitespace -_-

**Review by c2thorn — APPROVED** [review]

**Pull request merged**